syntax = "proto3";
package tum.mensa;

service MealRecommendationService {
// Recommend a meal for the user based on today's menu and previous favorites
rpc RecommendMeal (MealRecommendationRequest) returns (MealRecommendationResponse);
}

message MealRecommendationRequest {
string user_id = 1;
repeated string favorite_meals = 2;
repeated MenuMeal today_menu = 3;
}

message MenuMeal {
string name = 1;
string description = 2;
repeated string tags = 3; // e.g. ["vegetarian", "low-carb", "asian"]
}

message MealRecommendationResponse {
string recommended_meal_name = 1;
string reasoning = 2; // Optional: explain why this meal was selected
}